### PR TITLE
Misc updates/fixes: labkey.domain.createAndLoad fix for strictFieldValidation, labkey.webdav.put description support, add labkey.query.import

### PR DESCRIPTION
**Pull request by cnathe**
#### Rationale
Issue [43107](https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=43107): createAndLoad for Lists or Datasets using R returns an error
Issue [42950](https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=42950): Inserting file description using R API
Issue [42546](https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=42546): Improve performance of Rlabkey import for large number of rows

#### Related Pull Requests
* https://github.com/LabKey/labkey-api-r/pull/68
* https://github.com/LabKey/platform/pull/2265
* https://github.com/LabKey/testAutomation/pull/724

#### Changes
* Fix for labkey.domain.createAndLoad to set strictFieldValidation to false for study datasets and data classes
* Add param to labkey.webdav.put for setting the file description 
* Add new labkey.query.import command which uses file import to bulk import rows to a query/table
* Update labkey.domain.createAndLoad to use the new labkey.query.import command for the bulk data import

### CODE REVIEW

**Comment by cnathe** [comment]
Here is the Rlabkey doc info for the new `labkey.query.import` command:
Screen Shot 2021-05-18 at 11 50 05 AM
Screen Shot 2021-05-18 at 11 50 15 AM

**Comment by labkey-adam** [comment]
Hey @cnathe , were you planning to update the `labkey.setCurlOptions` docs in this PR, per https://github.com/LabKey/labkey-api-r/issues/67?

**Comment by cnathe** [comment]
> Hey @cnathe , were you planning to update the `labkey.setCurlOptions` docs in this PR, per #67?

@labkey-adam Thanks for the reminder. I have it on my list. I was planning to get these changes in with this PR and then do a follow-on mini-PR which actually did the version bump and release notes. I had planned to include that docs update in that PR but can just as easily add it here.

File: Rlabkey/R/labkey.domain.R [hunk 1]
```diff
@@ -241,7 +244,7 @@ labkey.domain.createAndLoad <- function(baseUrl=NULL, folderPath, name, descript
     labkey.domain.create(baseUrl = baseUrl, folderPath = folderPath, domainKind = domainKind,
         domainDesign = design, options = options)
 
-    labkey.insertRows(baseUrl = baseUrl, folderPath = folderPath, schemaName = schemaName, queryName= name, df)
+    labkey.query.import(baseUrl, folderPath, schemaName, queryName= name, toImport = df)
```
Comment (cnathe) [inline Rlabkey/R/labkey.domain.R:247]: @labkey-klum What do you think of this change? The new `labkey.query.import` call is much faster then the `labkey.insertRows` or `labkey.importRows` calls. Do you recall any specific reason why the `labkey.domain.createAndLoad` would need to use the `insertRows` or do you think it is fine to change it to `labkey.query.import`?

**Comment by cnathe** [comment]
Test plan for changes in this PR (note that the fix for Issue 43107 includes a small server side change against 21.3):

Issue 43107: createAndLoad for Lists or Datasets using R returns an error
- [x] verify that you can successfully use labkey.domain.createAndLoad for the following data types: datasets, list, sample type, data class
- [x] verify that the domain is created and that the data rows are loaded as expected

- [x] StudyDatasetVisit
``` 
labkey.domain.createAndLoad(
	baseUrl="http://localhost:8080/labkey", folderPath="Visit Based Study",
	name="Rlabkey dataset", df=dataset, domainKind="StudyDatasetVisit"
)
```
- [x] list
```
 labkey.domain.createAndLoad(
	baseUrl="http://localhost:8080/labkey", folderPath="Visit Based Study",
	name="Rlabkey list", df=dataset, domainKind="IntList", options=list(keyName="Key")
)
```
- [x] SampleSet
```
 labkey.domain.createAndLoad(
	baseUrl="http://localhost:8080/labkey", folderPath="Visit Based Study", 
	name="sampletype2", domainKind="SampleSet", df=df
)
```
- [x] DataClass
```
labkey.domain.createAndLoad(
	baseUrl="http://localhost:8080/labkey", folderPath="Visit Based Study", 
	name="dataclass2", domainKind="DataClass", df=df
)
```

Issue 42950: Inserting file description using R API
- [x] verify that the Rlabkey doc info for labkey.webdav.put has info on the new "description" param
- [x] verify that you can pass a "description" string value to the labkey.webdav.put call and that it shows as expected on the LK server side

Issue 42546: Improve performance of Rlabkey import for large number of rows
- [x] verify that the update Rlabkey package includes a new command for "labkey.query.import" and that the doc page for it looks as expected
- [x] verify the error messages for missing required params to labkey.query.import command make sense
- [x] verify usages of the labkey.query.import command without any options passed in
- [x] verify usages of the labkey.query.import command with options passed in (see examples below for sample type merge)
- [x] verify that the imported data looks as expected on the LK server side
- [x] verify that error message from import are shown as expected in R console (i.e. trying to insert duplicate key or a missing required field value)

```
labkey.query.import(
	"http://localhost:8080/labkey",
	folderPath="/Visit Based Study", schemaName="samples", queryName="sampletype",
	toImport=df, options = list(insertOption = "MERGE", auditBehavior = "DETAILED")
)
```

**Comment by cnathe** [comment]
And here is a copy of the Rlabkey package built by TeamCity that can be installed locally for testing.
[Rlabkey_2.6.0.tar.gz](https://github.com/LabKey/labkey-api-r/files/6503979/Rlabkey_2.6.0.tar.gz)

**Comment by XingY** [comment]
> And here is a copy of the Rlabkey package built by TeamCity that can be installed locally for testing.
> [Rlabkey_2.6.0.tar.gz](https://github.com/LabKey/labkey-api-r/files/6503979/Rlabkey_2.6.0.tar.gz)

Can you please also attach the doc pdf for testing?

File: Rlabkey/R/labkey.domain.R (hunk 1 above)
Comment (labkey-klum) [reply Rlabkey/R/labkey.domain.R:247]: yes, agreed that changing this to use the new `query.import` function will improve the perf on this function.

**Review by labkey-klum — APPROVED** [review]

**Pull request merged**